src/common.js

```diff
@@ -13,7 +14,10 @@ const Gtk = imports.gi.Gtk;
 ext.app_id = "org.gnome.Shell.Extensions.GSConnect";
 ext.app_path = "/org/gnome/Shell/Extensions/GSConnect";
 
-ext.metadata = JSON.parse(GLib.file_get_contents(ext.datadir + "/metadata.json")[1]);
+ext.metadata = JSON.parse(
+    ByteArray.toString(
+        GLib.file_get_contents(ext.datadir + "/metadata.json")[1]
+    ));
```

Comment:
Pretty much what should happen here is what gnome-shell is doing for byteArray:
```js
let metadata = GLib.file_get_contents(ext.datadir + "/metadata.json")[1];
gsconnect.metadata = JSON.parse(
    (metadata instanceof Uint8Array) ? imports.byteArray.toString(metadata) : metadata.toString()
);
```

Basically check each returned "bytearray" to see if it's a `Uint8Array` or not. Probably the backwards compatible version is faster since it's called automatically in the C++, but this is the only way to be backwards and forwards compatible I think.